azure-storage-blob/azure/storage/blob/blockblobservice.py

```diff
@@ -578,7 +578,7 @@ def create_blob_from_stream(
 
             # keep reading from stream util length of data >= count or reaching the end of stream
             while len(data) < count and len(data_chunk) is not 0:
-                data_chunk = stream.read(count)
+                data_chunk = stream.read(count - len(data))
```

Comment:
data_chunk is redundant, we can do:

```
data += stream.read(count - len(data))
```

Reply:
we use data_chunk to make sure len(data_chunk) is not 0(line 580)